src/test/java/com/mongodb/hibernate/jdbc/MongoConnectionTests.java

```diff
@@ -16,56 +16,297 @@
+                        mongoConnection.setAutoCommit(false);
+                    } else {
+                        assertThrows(SQLException.class, () -> mongoConnection.setAutoCommit(false));
+                    }
+                    verify(clientSession).startTransaction();
+                }
+
+                @ParameterizedTest(
+                        name = "AutoCommit changed (false -> true): commit existing transaction (successful: {0})")
+                @ValueSource(booleans = {true, false})
+                void testTryingToCommitExistingTransactionWhenAutoCommitChangedToTrue(boolean successful)
+                        throws SQLException {
+                    // given
+                    mongoConnection.setAutoCommit(false);
+                    verify(clientSession).startTransaction();
+
+                    assertFalse(mongoConnection.getAutoCommit());
+                    doReturn(true).when(clientSession).hasActiveTransaction();
+                    if (!successful) {
+                        doThrow(RuntimeException.class).when(clientSession).commitTransaction();
+                    }
+
+                    // when && then
+                    if (successful) {
+                        mongoConnection.setAutoCommit(true);
+                    } else {
+                        assertThrows(SQLException.class, () -> mongoConnection.setAutoCommit(true));
+                    }
+                    verify(clientSession).commitTransaction();
+                }
+
+                @Test
+                void testNoopWhenNoExistingTransactionAndAutoCommitChangedToTrue() throws SQLException {
+                    // given
+                    mongoConnection.setAutoCommit(false);
+                    verify(clientSession).startTransaction();
+                    mongoConnection.commit();
+                    verify(clientSession).commitTransaction();
+
+                    assertFalse(mongoConnection.getAutoCommit());
+
+                    // when
+                    mongoConnection.setAutoCommit(true);
+
+                    // then
+                    assertTrue(mongoConnection.getAutoCommit());
+                    verify(clientSession, times(2)).hasActiveTransaction();
+                    verifyNoMoreInteractions(clientSession);
+                }
+            }
+
+            @ParameterizedTest(
+                    name = "SQLException is thrown when setAutoCommit({0}) called on a closed MongoConnection")
+            @ValueSource(booleans = {true, false})
+            void testSQLExceptionThrowWhenCalledOnClosedConnection(boolean autoCommit) throws SQLException {
+                // given
+                mongoConnection.close();
+                verify(clientSession).close();
+
+                // when && then
+                assertThrows(SQLException.class, () -> mongoConnection.setAutoCommit(autoCommit));
+                verifyNoMoreInteractions(clientSession);
+            }
```

Comment:
Let's use the same approach when we call `verify(clientSession).close()` and `verifyNoMoreInteractions(clientSession)` in other similar tests:

- `TransactionIsolationLevelTests.testSQLExceptionThrownWhenConnectionClosed`
- `RollbackTests.testSQLExceptionThrownWhenConnectionClosedAlready`
- `CommitTests.testSQLExceptionThrownWhenConnectionClosedAlready`

Also, it seems that the method names of these tests should be the same, which is not the case. Descriptions should be similar, but currently, some of these tests have them, while others don't have them at all.